apps/veil/src/pages/portfolio/api/use-asset-prices.ts

```diff
@@ -23,18 +24,43 @@ export const useAssetPrices = (assets: Metadata[] = []) => {
   const { data, isLoading, error } = useQuery({
     queryKey: ['asset-prices', symbols],
     queryFn: async () => {
-      // If no assets, return empty array
       if (symbols.length === 0) {
         return [];
       }
 
-      // Fetch from summaries API
-      return apiFetch<SummaryData[]>('/api/summaries', {
-        search: '',
-        limit: '200',
-        offset: '0',
-        durationWindow: '1d',
+      // TODO: This is inefficient. A batch endpoint should be created to fetch prices for multiple assets at once.
+      // Fetch last candle for each symbol vs USDC to derive price.
+      const promises = symbols.map(async sym => {
+        if (sym.toUpperCase() === 'USDC') {
+          return {
+            baseAsset: { symbol: sym } as Metadata,
+            quoteAsset: { symbol: 'USDC' } as Metadata,
+            price: 1,
+          } as SummaryData;
+        }
+        try {
+          const candles = await apiFetch<CandleWithVolume[]>('/api/candles', {
+            baseAsset: sym,
+            quoteAsset: 'USDC',
+            durationWindow: '1h',
+          });
```

Comment:
note: with #2533, you could add `limit: 1` to improve the load time

Reply:
I'll make a note and add it